### PR TITLE
add few more assertions and handle some exceptions

### LINKED ISSUE #75: Handle the exceptions for missing arguments in commands

### PR DESCRIPTION
**Pull request by sliu107**
add few more assertions
handle the exceptions for missing argument(s) in some commands

### CODE REVIEW

File: src/main/java/seedu/tp/flashcard/EventFlashcard.java [hunk 1]
```diff
@@ -26,6 +26,7 @@ public EventFlashcard(String name, LocalDate startDate, LocalDate endDate, Strin
         assert !name.isEmpty() : "Invalid empty name!";
         assert startDate != null : "Invalid null startDate!";
         assert endDate != null : "Invalid null endDate!";
+        assert !startDate.isAfter(endDate) : "Invalid startDate and endDate";
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/flashcard/EventFlashcard.java]: Do we have checks for this when prompting user for input? If not I think we shouldn't make the whole program fail if the user enters a start date later than the end date.

File: src/main/java/seedu/tp/flashcard/PersonFlashcard.java [hunk 2]
```diff
@@ -27,8 +27,9 @@ public PersonFlashcard(String name, LocalDate birthDate, LocalDate deathDate, St
         assert !name.isEmpty() : "Invalid empty name!";
         assert birthDate != null : "Invalid null birthDate!";
         assert deathDate != null : "Invalid null deathDate!";
+        assert !birthDate.isAfter(deathDate) : "Invalid birthDate and deathDate";
```
Comment (Zhilin-Huang) [inline src/main/java/seedu/tp/flashcard/PersonFlashcard.java]: Same here

File: src/main/java/seedu/tp/flashcard/EventFlashcard.java (hunk 1 above)
Comment (quinnyyy) [reply src/main/java/seedu/tp/flashcard/EventFlashcard.java]: we can implement this for v2

**Review by quinnyyy — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/tp/flashcard/EventFlashcard.java (hunk 1 above)
Comment (sliu107) [reply src/main/java/seedu/tp/flashcard/EventFlashcard.java]: Yeah, I find there is not checking when prompting user for input. And I think maybe we can deal this with exceptions in v2.0 , just as Quinn advised.

Comment (sliu107) [reply src/main/java/seedu/tp/flashcard/EventFlashcard.java]: I think I can just delete these two assertion now.

**Review by quinnyyy — APPROVED** [review]
sounds good

**Pull request merged**